Source/WebCore/platform/graphics/cairo/PathCairo.cpp

```diff
@@ -479,66 +376,114 @@ void Path::applyIgnoringInlineData(const PathApplierFunction& function) const
         case CAIRO_PATH_MOVE_TO:
             pathElement.type = PathElement::Type::MoveToPoint;
             pathElement.points[0] = FloatPoint(data[1].point.x, data[1].point.y);
-            function(pathElement);
+            applier(pathElement);
             break;
+
         case CAIRO_PATH_LINE_TO:
             pathElement.type = PathElement::Type::AddLineToPoint;
             pathElement.points[0] = FloatPoint(data[1].point.x, data[1].point.y);
-            function(pathElement);
+            applier(pathElement);
             break;
+
         case CAIRO_PATH_CURVE_TO:
             pathElement.type = PathElement::Type::AddCurveToPoint;
             pathElement.points[0] = FloatPoint(data[1].point.x, data[1].point.y);
             pathElement.points[1] = FloatPoint(data[2].point.x, data[2].point.y);
             pathElement.points[2] = FloatPoint(data[3].point.x, data[3].point.y);
-            function(pathElement);
+            applier(pathElement);
             break;
+
         case CAIRO_PATH_CLOSE_PATH:
             pathElement.type = PathElement::Type::CloseSubpath;
-            function(pathElement);
+            applier(pathElement);
             break;
         }
     }
 }
 
-FloatRect Path::fastBoundingRectSlowCase() const
+bool PathCairo::isEmpty() const
 {
-    return boundingRect();
+    return !cairo_has_current_point(platformPath());
 }
 
-void Path::transform(const AffineTransform& transform)
+FloatPoint PathCairo::currentPoint() const
+{
+    // FIXME: Is this the correct way?
+    double x;
+    double y;
+    cairo_get_current_point(platformPath(), &x, &y);
+    return FloatPoint(x, y);
+}
+
+void PathCairo::transform(const AffineTransform& transform)
 {
     cairo_matrix_t matrix = toCairoMatrix(transform);
     cairo_matrix_invert(&matrix);
-    cairo_transform(ensureCairoPath(), &matrix);
+    cairo_transform(platformPath(), &matrix);
 
-    if (!m_elements)
-        return;
+    m_elementsStream = nullptr;
```

Comment:
The old code transformed `m_elements`. Should we do that here too? If not, why not?